### ISSUE TITLE
Recommended way to generate sequence-level embeddings with ESM C

### ISSUE THREAD

**Issue by pasley-cilantro** [evolutionaryscale/esm#162]
It seems that ESM C doesn't directly support generating sequence-level embeddings. What would be the recommended approach for creating these embeddings from `logits_output.embeddings`?

Should we use `logits_output.embeddings.mean(dim=1)` as-is, or would it be better to exclude the `<cls>` and `<eos>` tokens first?

Thank you!

**Comment by ebetica** [comment]
I believe the mean across the non bos/eos tokens should be the best way to get these embeddings. We have not done extensive tests here, but it would be a good place to start. If you finetune, you can try finetuning through just the cls token.
